src/gsy_e/models/strategy/energy_parameters/energy_params_eb.py

```diff
@@ -36,24 +36,273 @@
 
 if TYPE_CHECKING:
     from gsy_e.models.state import StateInterface
+    from gsy_e.models.area import AreaBase
+
+
+class BaseMarketEnergyParams(ABC):
+    """Base class for the energy parameters of specific markets."""
+    def __init__(self, posted_energy_kWh: DefaultDict):
+        self._posted_energy_kWh = posted_energy_kWh
+        self._area: Optional["AreaBase"] = None
+        self._profile_generator: Optional["ForwardTradeProfileGenerator"] = None
+
+    def activate(self, area: "AreaBase", profile_generator: "ForwardTradeProfileGenerator"):
+        """
+        Activate the energy parameters by providing extra arguments that are generated during the
+        activation process.
+        """
+        self._area = area
+        self._profile_generator = profile_generator
+
+    @abstractmethod
+    def get_posted_energy_kWh(self, market_slot: pendulum.DateTime) -> float:
+        """Get the already posted energy from the asset for this market slot."""
+
+    @abstractmethod
+    def increment_posted_energy(
+            self, market_slot: pendulum.DateTime, posted_energy_kWh: float):
+        """Increment the posted energy of the asset."""
+
+    @abstractmethod
+    def decrement_posted_energy(
+            self, market_slot: pendulum.DateTime, posted_energy_kWh: float):
+        """Decrement the posted energy of the asset."""
+
+    @abstractmethod
+    def get_available_load_energy_kWh(
+            self, market_slot: pendulum.DateTime, state: "LoadState",
+            peak_energy_kWh: float) -> float:
+        """Get the available (not traded) energy of a load asset."""
+
+    @abstractmethod
+    def get_available_pv_energy_kWh(
+            self, market_slot: pendulum.DateTime, state: "PVState",
+            peak_energy_kWh: float) -> float:
+        """Get the available (not traded) energy of a PV asset."""
```

Comment:
I still find the `available_load_energy` wording somehow confusing. The reason is that we've always used "require/desired energy" for loads' energy, while the "available energy" naming was always related to production assets. Do you think there can be a different way to name this?

On another side: we had previously structured the energy parameter classes to be grouped into consumption vs production. Now we're mixing these two, by adding both load and PV methods in the same class. Despite the name, the new `*EnergyParams` classes don't have the same interface as the previous `*EnergyParams` classes, and `state` isn't an instance variable anymore. So I'm not sure about how these new classes will be used. 
Are they only supposed to be used in the `ForwardEnergyParams._forward_energy_params` dictionary (so we will just keep using `Production/ConsumptionStandardProfileEnergyParameters` as before)? Or are they also going to be used "directly"?
In the first case, I'd suggest adding a note in their docstrings stating something like "This class should not be instantiated on its own. Please only use `Production/ConsumptionStandardProfileEnergyParameters`". What do you think?

Reply:
> I still find the available_load_energy wording somehow confusing. The reason is that we've always used "require/desired energy" for loads' energy, while the "available energy" naming was always related to production assets. Do you think there can be a different way to name this?

We can harmonise the names to something like `required_load_energy` and `available_pv_energy` in order to be consistent with the state classes, but I frankly would prefer using a common name for both, because they are both going to be used for the same purpose: get the energy that is available for trading. Another possibility would be to name both `non_traded_energy`, do you think that this would make it more easily readable ?